lib/bundler/settings/validator.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Bundler
+  class Settings
+    class Validator
+      class Rule
+        attr_reader :description
+
+        def initialize(keys, description, &validate)
+          @keys = keys
+          @description = description
+          @validate = validate
+        end
+
+        def validate(key, value, settings)
+          instance_exec(key, value, settings, &@validate)
```

Comment:
I've always been partial to the syntax `@validate.call(key, value, settings)` using a proc. I find it easier to parse than `exec`s.